### PR TITLE
Fix rate-limiting

### PR DESCRIPTION
**Pull request by MuriloDalRi**
Dependapanda has stopped notifying Slack since the end of March.

It seems like Github is being more aggressive in rate-limiting requests to their API.

This PR attempts to fix this by making API requests more spaced.

[Trello](https://trello.com/c/NpeMpcep/2886-fix-dependapanda-again-5)

### CODE REVIEW

File: lib/gateways/pull_request.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ def execute
     end
 
     def approved_pull_requests
-      query = "is:pr user:alphagov state:open author:app/dependabot author:app/dependabot-preview review:approved"
+      query = "is:pr user:alphagov state:open author:app/dependabot review:approved"
```
Comment (ChrisBAshton) [inline lib/gateways/pull_request.rb]: Nice improvement 👍 
Nitpick: could you link to some evidence in the commit message?

File: Rakefile [hunk 2]
```diff
@@ -45,19 +45,32 @@ end
...
 desc "Fetch all of the data we need from github in a slow way, and write to the cache"
 task :warm_the_cache do
+  puts "Fetching repos..."
   Rake::Task["fetch_repos"].invoke
-  puts "fetched repos"
-  sleep 10
+  puts "Finished fetching repos."
+  puts "Sleeping for 60 seconds to avoid rate-limiting"
+
+  sleep 60
```
Comment (ChrisBAshton) [inline Rakefile:53]: Might be worth moving this out to a `def avoid_rate_limit` as we use it a few times.

File: spec/acceptance/app_spec.rb [hunk 3]
```diff
@@ -220,7 +220,7 @@ def app
 
   context "Stats page" do
     before do
-      stub_request(:get, "https://api.github.com/search/issues?q=is:pr%20user:alphagov%20author:app/dependabot")
+      stub_request(:get, "https://api.github.com/search/issues?q=is:pr%20user:alphagov%20state:open%20author:app/dependabot%20archived:false")
```
Comment (ChrisBAshton) [inline spec/acceptance/app_spec.rb:223]: Can you explain (in the commit message) why these changes are necessary? Does the Octokit SDK now append these extra parameters to requests?

File: README.md [hunk 4]
```diff
@@ -38,7 +38,6 @@ Running this will start your application at [localhost:9292](localhost:9292)
 ### Environment variables
 
 - `GITHUB_TOKEN` - OAuth token generated on GitHub which does not require any special permissions
-  - Used to interact with the GitHub API, although not required it will help avoid limiting
```
Comment (ChrisBAshton) [inline README.md:41]: Good call 👍

File: lib/gateways/pull_request.rb [hunk 5]
```diff
@@ -14,7 +14,7 @@ def execute
     end
 
     def approved_pull_requests
-      query = "is:pr user:alphagov state:open author:app/dependabot review:approved"
+      query = "is:pr user:alphagov state:open author:app/dependabot archived:false review:approved"
```
Comment (ChrisBAshton) [inline lib/gateways/pull_request.rb:17]: Ah - I'm guessing the "Fix tests" commit is because of these changes. Can you squash them together?

**Review by ChrisBAshton — COMMENTED** [review]
Looking good 👍  has this been tested? Or is that blocked on the [GitHub/Heroku security incident fallout](https://github.blog/2022-04-15-security-alert-stolen-oauth-user-tokens/) and/or the fact that some of our Heroku stacks are EOL?

File: lib/gateways/pull_request.rb (hunk 5 above)
Comment (MuriloDalRi) [reply lib/gateways/pull_request.rb:17]: Done

File: Rakefile (hunk 2 above)
Comment (MuriloDalRi) [reply Rakefile]: I tried that but Rubocop wasn't happy

File: lib/gateways/pull_request.rb (hunk 1 above)
Comment (MuriloDalRi) [reply lib/gateways/pull_request.rb]: I added a link to a PR that has a bit more info

**Comment by MuriloDalRi** [comment]
> Looking good 👍 has this been tested? Or is that blocked on the [GitHub/Heroku security incident fallout](https://github.blog/2022-04-15-security-alert-stolen-oauth-user-tokens/) and/or the fact that some of our Heroku stacks are EOL?



> Looking good 👍 has this been tested? Or is that blocked on the [GitHub/Heroku security incident fallout](https://github.blog/2022-04-15-security-alert-stolen-oauth-user-tokens/) and/or the fact that some of our Heroku stacks are EOL?

I've manually pushed this to Heroku and also tested it locally and it seems to be working. I think I've addressed all comments now if you want to have another look

**Review by ChrisBAshton — APPROVED** [review]
LGTM 👍 great job!
